### ISSUE TITLE
Anomaly witness with STATES and UPDATE ROW in Safari vs Chrome

### ISSUE THREAD

**Issue by raventechnologiesinc** [Budibase/budibase#13109]
**Checklist**
- [X ] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**
  - Tenant ID: ship2vietnam.budibase.app -->

**Describe the bug**
Today I witness something that is unexpainable.

Our app use a STATE to store userId and Filters parameters for table display.

In chrome everything works fines.

One of our used is on Safari and reported an issue.
They were seeing data UNFILTERED, meaning the STATE for one of our filters did not take.
Are STATES stored in COOKIES?

They also noticed that UPDATING records did not take.
We concurrently tested with Chrome simultaneously.
UPDATE ROW commands took, MySQL updated and table displayed properly.

Mean while on Safari it did not (This is impossible!)
A database sql call has nothing to do with the interface.

So clearing cache.  Re login to get STATES updated.
got the tables to filter properly.  

Is this a ghost in the machine.

If STATES are stored as COOKIES and not server side, is the only thing that can cause this.

**Desktop (please complete the following information):**
 - WIN11
 - Chrome safari edge

**Comment by melohagan** [comment]
> Are STATES stored in COOKIES?

App state is stored in [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage)

**Comment by melohagan** [comment]
> Is this a ghost in the machine.

![image](https://github.com/Budibase/budibase/assets/101575380/be90fb0b-d7f0-4f78-bb5e-a8f6c2e2d9e3)

Probably not.

**Comment by melohagan** [comment]
@raventechnologiesinc Are you able to provide an app export, and some screenshots to highlight the problem please?

**Comment by raventechnologiesinc** [comment]
> @raventechnologiesinc Are you able to provide an app export, and some screenshots to highlight the problem please?

[SHIPTOVIETNAM-export-1708569754720.tar.gz](https://github.com/Budibase/budibase/files/14381213/SHIPTOVIETNAM-export-1708569754720.tar.gz)

1) Th update row not taking was concluded to the be the date picker bug.  The current data was 21/Feb.  Picker date was 21/Feb, user kept changing to 22/Feb.  App kept updating DB as 21/Feb. A few hours rolled back.  User reported data not updating.  But it was a timedate anomaly.

2) The STATE issue. in app 'newbillright'.  We used a persistant state to filter user class and display their records only.  Everything works fine if we store that STATE from the 'login' page.  But opening 'newbillright' on a virgin browser without ever logging in.  User can see all records without login/ setting a STATE.  That's why we suspected STATES were being stored as COOKIES.

Now we suspect code not catching and  undefined/null/blank STATE variable.  Give our Data provider filter values of blank/undefined/null  allowing users to see each other's records.

**Comment by melohagan** [comment]
Hey @raventechnologiesinc Thanks for the app export. Which screen should I be looking at? Can you also provide some screenshots as well please to show the problem?

**Comment by raventechnologiesinc** [comment]
/newbillright

Originally we checked for STATE = '' , to force redirect app to login screen.
This version when on Safari it did not understand and allowed full access to our app.
(We believe this is a JS coding error, not a BB bug)

Upon logout user we SET the STATE to empty.
(Found bug)

Virgin public users on Safari get full access. 
(Found bug)

Now we CLEAR the STATE.  Compare all empty states to be sure.

(We use a simple hack for authentication during development)
LOGIN sets a persistant state, LOGOUT set state to empty/blank field.

We solved the issue by explicitly catching STATES in all states.

...STATE = null , STATE = undefined, STATE = ''

Allowing for all possibilities Across ALL browsers.


A built in JS debugger would be so helpful with BB.
